Can't resolve implicit conversion in 0.23 but success in 0.22

## minimized code

+ Position.scala
```Scala
import scala.quoted.{Expr, QuoteContext}

case class Position()
object Position {
  implicit inline def here: Position = ${ genPosition }

  private def genPosition(given Quotes): Expr[Position] = {
    import quotes.reflect._
    // interesting that if we remove the following line => compile successfully
    val lineNo: Int = Position.ofMacroExpansion.startLine
    '{ Position() }
  }
}
```
+ Test.scala
```scala
import scala.concurrent.duration.FiniteDuration
import scala.language.implicitConversions
import scala.reflect.ClassTag

class AnyShouldWrapper[T]

class Test {
  implicit def convertToAnyShouldWrapper[T](o: T)(implicit pos: Position): AnyShouldWrapper[T] = ???

  def expectMsgType[T](implicit t: ClassTag[T]): T = ???
  def expectMsgType[T](max: FiniteDuration)(implicit t: ClassTag[T]): T = ???

  def test(): Unit = {
    expectMsgType[Int]: AnyShouldWrapper[Int]
  }
}
```

## Compilation output for dotty `0.23.0-RC1`

```scala
[error] -- [E134] Type Mismatch Error: /Users/thanhbv/ohze/dotty-example/src/main/scala/Test.scala:14:4 
[error] 14 |    expectMsgType[Int]: AnyShouldWrapper[Int]
[error]    |    ^^^^^^^^^^^^^
[error]    |None of the overloaded alternatives of method expectMsgType in class Test with types
[error]    | [T](max: concurrent.duration.FiniteDuration)(implicit t: reflect.ClassTag[T]): T
[error]    | [T](implicit t: reflect.ClassTag[T]): T
[error]    |match type arguments [Int] and expected type AnyShouldWrapper[Int]
```

## expectation
compile successfully as with dotty `0.22.0-RC1`

If I change the line `val lineNo: Int = rootPosition.startLine` to
```scala
    try rootPosition.startLine catch {
      case e => e.printStackTrace()
    }
```
=> output:
<details>

```scala
java.lang.AssertionError: assertion failed
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
	at dotty.tools.dotc.util.Spans$Span$.start$extension(Spans.scala:44)
	at dotty.tools.dotc.util.SourcePosition.start(SourcePosition.scala:47)
	at dotty.tools.dotc.util.SourcePosition.startLine(SourcePosition.scala:48)
	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.Position_startLine(ReflectionCompilerInterface.scala:1601)
	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.Position_startLine(ReflectionCompilerInterface.scala:1601)
	at scala.tasty.Reflection$positionOps$.startLine(Reflection.scala:2685)
	at Position$.genPosition(Position.scala:10)
	at Position$.inline$genPosition(Position.scala:8)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at dotty.tools.dotc.transform.Splicer$Interpreter.interpretedStaticMethodCall$$anonfun$2$$anonfun$1(Splicer.scala:320)
	at dotty.tools.dotc.transform.Splicer$Interpreter.stopIfRuntimeException(Splicer.scala:380)
	at dotty.tools.dotc.transform.Splicer$Interpreter.interpretedStaticMethodCall$$anonfun$1(Splicer.scala:320)
	at dotty.tools.dotc.transform.Splicer$Interpreter.interpretTree(Splicer.scala:245)
	at dotty.tools.dotc.transform.Splicer$Interpreter.interpretTree$$anonfun$4(Splicer.scala:265)
	at dotty.tools.dotc.transform.Splicer$.$anonfun$2(Splicer.scala:49)
	at scala.Option.fold(Option.scala:263)
	at dotty.tools.dotc.transform.Splicer$.splice(Splicer.scala:49)
	at dotty.tools.dotc.typer.Inliner.dotty$tools$dotc$typer$Inliner$$expandMacro(Inliner.scala:1282)
	at dotty.tools.dotc.typer.Inliner$InlineTyper.typedApply(Inliner.scala:1109)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2208)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:123)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.ReTyper.typedTyped(ReTyper.scala:60)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2213)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:123)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2300)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Inliner.op$1(Inliner.scala:623)
	at dotty.tools.dotc.typer.Inliner.inlined(Inliner.scala:642)
	at dotty.tools.dotc.typer.Inliner$.inlineCall(Inliner.scala:115)
	at dotty.tools.dotc.typer.Typer.adaptNoArgsOther$5(Typer.scala:2939)
	at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:3036)
	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3244)
	at dotty.tools.dotc.typer.Typer.op$3(Typer.scala:2635)
	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:2636)
	at dotty.tools.dotc.typer.Implicits.op$2(Implicits.scala:1379)
	at dotty.tools.dotc.typer.Implicits.typedImplicit(Implicits.scala:1423)
	at dotty.tools.dotc.typer.Typer.typedImplicit(Typer.scala:83)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.tryImplicit(Implicits.scala:1458)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.rank$1(Implicits.scala:1538)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicits(Implicits.scala:1631)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1652)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1660)
	at dotty.tools.dotc.typer.Implicits.op$1(Implicits.scala:1331)
	at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1367)
	at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:83)
	at dotty.tools.dotc.typer.Implicits.inferImplicitArg(Implicits.scala:1112)
	at dotty.tools.dotc.typer.Typer.inferImplicitArg(Typer.scala:83)
	at dotty.tools.dotc.typer.Typer.implicitArgs$2(Typer.scala:2754)
	at dotty.tools.dotc.typer.Typer.addImplicitArgs$4(Typer.scala:2777)
	at dotty.tools.dotc.typer.Typer.adaptNoArgsImplicitMethod$2(Typer.scala:2844)
	at dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:3014)
	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3244)
	at dotty.tools.dotc.typer.Typer.op$3(Typer.scala:2635)
	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:2636)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Implicits.tryConversion$1(Implicits.scala:1394)
	at dotty.tools.dotc.typer.Implicits.op$2(Implicits.scala:1406)
	at dotty.tools.dotc.typer.Implicits.typedImplicit(Implicits.scala:1423)
	at dotty.tools.dotc.typer.Typer.typedImplicit(Typer.scala:83)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.tryImplicit(Implicits.scala:1458)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.rank$1(Implicits.scala:1538)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicits(Implicits.scala:1631)
	at dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1652)
	at dotty.tools.dotc.typer.Implicits.op$1(Implicits.scala:1331)
	at dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1367)
	at dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:83)
	at dotty.tools.dotc.typer.Implicits.inferView(Implicits.scala:683)
	at dotty.tools.dotc.typer.Typer.inferView(Typer.scala:83)
	at dotty.tools.dotc.typer.Implicits.viewExists(Implicits.scala:658)
	at dotty.tools.dotc.typer.Typer.viewExists(Typer.scala:83)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.isCompatible(ProtoTypes.scala:37)
	at dotty.tools.dotc.typer.Typer.isCompatible(Typer.scala:83)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.testCompat$1(ProtoTypes.scala:45)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.normalizedCompatible$$anonfun$1(ProtoTypes.scala:59)
	at dotty.tools.dotc.core.Contexts$Context.test$$anonfun$1(Contexts.scala:322)
	at dotty.tools.dotc.core.TyperState.test(TyperState.scala:116)
	at dotty.tools.dotc.core.Contexts$Context.test(Contexts.scala:322)
	at dotty.tools.dotc.typer.ProtoTypes$Compatibility.normalizedCompatible(ProtoTypes.scala:59)
	at dotty.tools.dotc.typer.Typer.normalizedCompatible(Typer.scala:83)
	at dotty.tools.dotc.typer.Applications.$anonfun$32(Applications.scala:1780)
	at scala.collection.immutable.List.filter(List.scala:487)
	at dotty.tools.dotc.typer.Applications.op$2(Applications.scala:1780)
	at dotty.tools.dotc.typer.Applications.resolveOverloaded1(Applications.scala:1845)
	at dotty.tools.dotc.typer.Applications.resolve$1(Applications.scala:1606)
	at dotty.tools.dotc.typer.Applications.resolveOverloaded(Applications.scala:1645)
	at dotty.tools.dotc.typer.Typer.resolveOverloaded(Typer.scala:83)
	at dotty.tools.dotc.typer.Applications.resolveMapped(Applications.scala:1862)
	at dotty.tools.dotc.typer.Typer.resolveMapped(Typer.scala:83)
	at dotty.tools.dotc.typer.Applications.op$2(Applications.scala:1774)
	at dotty.tools.dotc.typer.Applications.resolveOverloaded1(Applications.scala:1845)
	at dotty.tools.dotc.typer.Applications.resolve$1(Applications.scala:1606)
	at dotty.tools.dotc.typer.Applications.resolveOverloaded(Applications.scala:1645)
	at dotty.tools.dotc.typer.Typer.resolveOverloaded(Typer.scala:83)
	at dotty.tools.dotc.typer.Typer.adaptOverloaded$1(Typer.scala:2659)
	at dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:3218)
	at dotty.tools.dotc.typer.Typer.op$3(Typer.scala:2635)
	at dotty.tools.dotc.typer.Typer.adapt(Typer.scala:2636)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2424)
	at dotty.tools.dotc.typer.Applications.typedFunPart$$anonfun$1(Applications.scala:820)
	at dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:2432)
	at dotty.tools.dotc.typer.Applications.typedFunPart(Applications.scala:828)
	at dotty.tools.dotc.typer.Typer.typedFunPart(Typer.scala:83)
	at dotty.tools.dotc.typer.Applications.typedTypeApply(Applications.scala:1014)
	at dotty.tools.dotc.typer.Typer.typedTypeApply(Typer.scala:83)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2226)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.ascription$1(Typer.scala:695)
	at dotty.tools.dotc.typer.Typer.typedTyped$$anonfun$4(Typer.scala:733)
	at dotty.tools.dotc.typer.Typer.cases$1(Typer.scala:684)
	at dotty.tools.dotc.typer.Typer.typedTyped(Typer.scala:734)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2213)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2424)
	at dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:826)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2216)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2424)
	at dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:1742)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2195)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2265)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2338)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2382)
	at dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:1875)
	at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2198)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2265)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2338)
	at dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:2382)
	at dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2001)
	at dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2239)
	at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2266)
	at dotty.tools.dotc.typer.Typer.op$1(Typer.scala:2304)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2313)
	at dotty.tools.dotc.typer.Typer.typed(Typer.scala:2316)
	at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:2424)
	at dotty.tools.dotc.typer.FrontEnd.liftedTree1$2(FrontEnd.scala:78)
	at dotty.tools.dotc.typer.FrontEnd.typeCheck$$anonfun$1(FrontEnd.scala:83)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.typer.FrontEnd.monitor(FrontEnd.scala:42)
	at dotty.tools.dotc.typer.FrontEnd.typeCheck(FrontEnd.scala:84)
	at dotty.tools.dotc.typer.FrontEnd.runOn$$anonfun$3(FrontEnd.scala:114)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.immutable.List.foreach(List.scala:305)
	at dotty.tools.dotc.typer.FrontEnd.runOn(FrontEnd.scala:114)
	at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:167)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
	at dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
	at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at dotty.tools.dotc.Run.runPhases$5(Run.scala:177)
	at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:185)
	at dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:64)
	at dotty.tools.dotc.Run.compileUnits(Run.scala:192)
	at dotty.tools.dotc.Run.compileSources(Run.scala:129)
	at dotty.tools.dotc.Run.compile(Run.scala:112)
	at dotty.tools.dotc.Driver.doCompile(Driver.scala:36)
	at dotty.tools.dotc.Driver.process(Driver.scala:189)
	at dotty.tools.dotc.Main.process(Main.scala)
	at xsbt.CachedCompilerImpl.run(CachedCompilerImpl.java:69)
	at xsbt.CompilerInterface.run(CompilerInterface.java:41)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:248)
	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:122)
	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:95)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:186)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3(MixedAnalyzingCompiler.scala:82)
	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3$adapted(MixedAnalyzingCompiler.scala:77)
	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:215)
	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:77)
	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:146)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:343)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:343)
	at sbt.internal.inc.Incremental$.doCompile(Incremental.scala:120)
	at sbt.internal.inc.Incremental$.$anonfun$compile$4(Incremental.scala:100)
	at sbt.internal.inc.IncrementalCommon.recompileClasses(IncrementalCommon.scala:180)
	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:98)
	at sbt.internal.inc.Incremental$.$anonfun$compile$3(Incremental.scala:102)
	at sbt.internal.inc.Incremental$.manageClassfiles(Incremental.scala:155)
	at sbt.internal.inc.Incremental$.compile(Incremental.scala:92)
	at sbt.internal.inc.IncrementalCompile$.apply(Compile.scala:75)
	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:348)
	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:301)
	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:168)
	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:248)
	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:74)
	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:1762)
	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:1735)
	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
	at sbt.std.Transform$$anon$4.work(Transform.scala:67)
	at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
	at sbt.Execute.work(Execute.scala:290)
	at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
(repeat 3 times)
</details>

--- a/tests/pos/i8581/Macro_1.scala
+++ b/tests/pos/i8581/Macro_1.scala
@@ -1,0 +1,11 @@
+import scala.quoted.*
+
+case class Position()
+object Position {
+  implicit inline def here: Position = ${ genPosition }
+
+  private def genPosition(using Quotes): Expr[Position] = {
+    val lineNo: Int = quotes.reflect.Position.ofMacroExpansion.startLine
+    '{ Position() }
+  }
+}

--- a/tests/pos/i8581/Test_2.scala
+++ b/tests/pos/i8581/Test_2.scala
@@ -1,0 +1,16 @@
+import scala.concurrent.duration.FiniteDuration
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+class AnyShouldWrapper[T]
+
+class Test {
+  implicit def convertToAnyShouldWrapper[T](o: T)(implicit pos: Position): AnyShouldWrapper[T] = ???
+
+  def expectMsgType[T](implicit t: ClassTag[T]): T = ???
+  def expectMsgType[T](max: FiniteDuration)(implicit t: ClassTag[T]): T = ???
+
+  def test(): Unit = {
+    expectMsgType[Int]: AnyShouldWrapper[Int]
+  }
+}

